src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -959,6 +959,20 @@ void RaftReplDev::handle_commit(repl_req_ptr_t rreq, bool recovery) {
     if (!rreq->is_proposer()) { rreq->clear(); }
 }
 
+void RaftReplDev::handle_config_commit(const repl_lsn_t lsn, raft_cluster_config_ptr_t& new_conf) {
+    // when reaching here, the new config has already been applied to the cluster.
+    // since we didn't create repl req for config change, we just need to update m_commit_upto_lsn here.
+
+    // keep this variable in case it is needed later
+    (void) new_conf;
+    auto prev_lsn = m_commit_upto_lsn.load(std::memory_order_relaxed);
+    RD_DBG_ASSERT_GT(lsn, prev_lsn,
+                     "Out of order commit of lsns, it is not expected in RaftReplDev. cur_lsns={}, prev_lsns={}",
+                     lsn, prev_lsn);
+    RD_DBG_ASSERT(m_commit_upto_lsn.compare_exchange_strong(prev_lsn, lsn),
```

Comment:
not sure whether it will take effect if in release or releaseWithDebug mode.  will it only work when debug mode? pls check this.

if we want to guarantee `m_commit_upto_lsn.compare_exchange_strong(prev_lsn, lsn) ` is called, pls use a separate line for this code.

Reply:
it will be execute as normal
```
#ifndef NDEBUG
#define DEBUG_ASSERT(cond, msg, ...) RELEASE_ASSERT(cond, msg, ##__VA_ARGS__)
#else
#define DEBUG_ASSERT(cond, msg, ...)
```